KubeArmor/common/common.go

```diff
@@ -443,3 +444,100 @@ func WriteToFile(val interface{}, destFile string) error {
 	}
 	return nil
 }
+
+// ============================== //
+// == Check for Empty policies == //
+// ============================= //
+
+// Check For Empty Container Policy Function
+
+func CheckForEmptyContainerPolicy(policyEvent tp.K8sKubeArmorPolicyEvent)(EmptyPolicy bool){
+
+	EmptyPolicy = true
+	MetaDataType := reflect.TypeOf(policyEvent.Object.Metadata)
+	MetaDataValue := reflect.ValueOf(policyEvent.Object.Metadata)
+
+
+	for i := 0; i < MetaDataType.NumField() && EmptyPolicy; i++ {
+
+		struct_value := MetaDataValue.FieldByName(MetaDataType.Field(i).Name)
+	
+		switch MetaDataType.Field(i).Type.Kind() {
+
+		case reflect.Array, reflect.Map, reflect.Slice:
+
+			if struct_value.Len() != 0{
+				EmptyPolicy=false
+			}
+		case reflect.String:
+
+			if struct_value.String() != ""{
+				EmptyPolicy=false
+			}
+
+		case reflect.Int:
+
+			if struct_value.Int() != 0{
+
+				EmptyPolicy=false
+			}
+		case reflect.Ptr:
+
+			if !struct_value.IsNil(){	
+
+				EmptyPolicy=false
+			
+			}
+		}
+	}
+
+	return EmptyPolicy
+}
+
+
+// Check For Empty Host Policy Function
+
+func CheckForEmptyHostPolicy(policyEvent tp.K8sKubeArmorHostPolicyEvent)(EmptyPolicy bool){
+
+	EmptyPolicy = true
+	MetaDataType := reflect.TypeOf(policyEvent.Object.Metadata)
+	MetaDataValue := reflect.ValueOf(policyEvent.Object.Metadata)
+
+
+	for i := 0; i < MetaDataType.NumField() && EmptyPolicy; i++ {
+
+		struct_value := MetaDataValue.FieldByName(MetaDataType.Field(i).Name)
+	
+		switch MetaDataType.Field(i).Type.Kind() {
+
+		case reflect.Array, reflect.Map, reflect.Slice:
+
+			if struct_value.Len() != 0{
+				EmptyPolicy=false
+			}
+		case reflect.String:
+
+			if struct_value.String() != ""{
+				EmptyPolicy=false
+			}
+
+		case reflect.Int:
+
+			if struct_value.Int() != 0{
+
+				EmptyPolicy=false
+			}
+		case reflect.Ptr:
+
+			if !struct_value.IsNil(){	
+
+				EmptyPolicy=false
+			
+			}
+		}
+	}
```

Comment:
Why do we need to use the reflect package here?

Shouldn't the check be as simple as 
`if policyEvent.Object.Metadata &&  policyEvent.Object.Metadata.Name != "" {}`

Reply:
Okay I saw that the Name field in Metadata is optional. Actually there is another field Generate_Name that one can use to generate metadata name. So a valid policy can do without that.

![image](https://user-images.githubusercontent.com/59079323/191537882-b4b4a242-34fa-4c5e-8332-e079e5fe5dac.png)

Reply:
Thanks. You are right even if that were to be the case, then I should check for only Name and GenerateName field. Just when I looked at it I saw that all the fields were optional and perhaps it needed just one field.

Reply:
> Is required when creating resources, some resources **may allow** a client to request the generation of an appropriate  automatically.

As it mentions may request client for auto generation, KubeArmor doesn't do that, so Ig we can deem it as required.

Reply:
Okay thanks a lot. I will implement the change